### PR TITLE
Refactor generate groups

### PR DESCRIPTION
**Pull request by marshallds**
## Migrations
YES

## Description
This simplifies and improves the clarity of how matching works. The migrations remove fields that no longer do anything and were confusing.

## Definition of Done

- [x] This PR has appropriate test coverage.

### CODE REVIEW

**Review by aelydens — APPROVED** [review]
Looks good to me.

I'm far enough removed from this project that my review of these PRs is unnecessary, tbh -- you guys have a much more up-to-date understanding of direction and priorities. You shouldn't feel like you have to have me review moving forward; I trust you to take this in the direction you feel is most important. You're doing a great job with this project, and thanks for taking it on!

**Review by spogburn — APPROVED** [review]
It looks fine to me.

**Comment by rebeccafae** [comment]
Thanks for reviewing @spogburn and @aelydens 😄 
I always like to have lots of eyes when we're making big changes like this one

File: app/models/peer_group.rb [hunk 1]
```diff
@@ -17,53 +13,39 @@ class PeerGroup < ActiveRecord::Base
...
+      users.sample(group_size).tap do |selected|
+        users -= selected
+      end
     end
   end
 
-  def self.update_users!(group)
-    group.each { |user| user.update(is_assigned_peer_group: true) }
+  def send_mail
+    SlackNotification.notify(:peer_group_send_mail, "Users ##{users.map(&:id).join(", #")}")
```
Comment (rebeccafae) [inline app/models/peer_group.rb:48]: I wonder if we want to extract the message to make this more readable

**Review by rebeccafae — APPROVED** [review]

**Pull request merged**